src/middlewares.js

```diff
@@ -105,8 +112,16 @@ export function handleParseHeaders(req, res, next) {
         info.masterKey = req.body._MasterKey;
         delete req.body._MasterKey;
       }
-      if (req.body._context && req.body._context instanceof Object) {
-        info.context = req.body._context;
+      if (req.body._context) {
+        if (req.body._context instanceof Object) {
+          info.context = req.body._context;
+        } else {
+          try {
+            info.context = JSON.parse(req.body._context);
```

Comment:
- Should we really do force-parsing? If the developer submits anything other than an object, it is parsed, beyond the dev's control. I am not sure that is wanted in every case.
- I think we should make context to be of type Object mandatorily. Otherwise we complicate the context handling in the sever code, as we never now what type it actually is. With this change it would be possible to parse `1` or `false` instead of an object.

Reply:
I've added a check to make sure context is always an object.

Your concern holds true for `header` as well. The problem is some encoders only encode to JSON strings, like the Swift SDK, but to make sure `context` is always an object I've added a check. If the only way to send `context` was through `_context` in the body, the change would be necessary to support the Swift SDK and those alike.

Let me know what you think and if you see any other loop holes.

Reply:
`instanceof` and `typeof` produce false positives, use this to identify a JS object literal:

```
Object.prototype.toString.call(info.context) === '[object Object]'
```

Reply:
Made these changes